syntax = "proto3";

message Model {
    repeated int32 shape = 1;
    ChannelOpts channel_opts = 2;
    repeated DTree classifier = 3;
    repeated float theta = 4;
}

message ChannelOpts {
    int32 shrink = 1;
    int32 n_per_oct = 2;
    int32 smooth = 3;
    string target_dtype = 4;
    repeated string func = 5;
}

message DTree {
    repeated int32 feature = 1;
    repeated float threshold = 2;
    repeated int32 left = 3;
    repeated int32 right = 4;
    repeated float prediction = 5;
}